### PR TITLE
Fix flaky not working for function tests under nose.

### LINKED ISSUE #22: crash if using Nose short test naming convention
If a test uses Nose's "short style" tests, Flaky gets confused about the test name, and crashes.

Test case:

test_nose_example.py
-------------------------------
```
# USAGE:
# 	nosetests -s --with-flaky ./test/test_nose_example.py

from nose.tools import eq_

def test1():
    eq_(1, 1)
```

### PR DESCRIPTION
**Pull request by Jeff-Meadows**
Fixes #22.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

File: flaky/flaky_nose_plugin.py [hunk 1]
```diff
@@ -157,22 +157,26 @@ def _get_test_method_name(test):
...
             :class:`nose.case.Test`
         :return:
-            The test method (and its name) that is being run by the test
+            The test callable (and its name) that is being run by the test
         :rtype:
             `tuple` of `callable`, `unicode`
         """
-        method_name = cls._get_test_method_name(test)
-        return getattr(test.test, method_name), method_name
+        method_name = cls._get_test_callable_name(test)
```
Comment (jmoldow) [inline flaky/flaky_nose_plugin.py]: Should this be ``test_name``?

File: flaky/_flaky_plugin.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def _handle_test_error_or_failure(self, test, err):
             `bool`
         """
         try:
-            test_method, test_method_name = self._get_test_method_and_name(
+            test_method, test_method_name = self._get_test_callable_and_name(
```
Comment (jmoldow) [inline flaky/_flaky_plugin.py]: There are still lots of ``_method`` variables.

File: flaky/flaky_nose_plugin.py (hunk 1 above)
Comment (Jeff-Meadows) [inline flaky/flaky_nose_plugin.py]: Yes. Good catch.

File: flaky/flaky_pytest_plugin.py [hunk 3]
```diff
@@ -171,34 +175,34 @@ def terminal_summary(self, stream):
...
             The test that has raised an error or succeeded
         :type test:
             :class:`Function`
         :return:
-            The name of the test method that is being run by the test
+            The name of the test callable that is being run by the test
         :rtype:
             `unicode`
         """
         return test.name
```
Comment (jmoldow) [inline flaky/flaky_pytest_plugin.py]: Just checking, is this still the correct thing to do?

File: test/test_example.py [hunk 4]
```diff
@@ -65,3 +65,7 @@ def test_flaky_thing_that_fails_then_succeeds(self):
         self._threshold += 1
         if self._threshold < 1:
             raise Exception("Threshold is not high enough.")
+
+
+def test_function():
```
Comment (jmoldow) [inline test/test_example.py]: Do we need to apply ``@flaky``? Or does this serve our purposes no matter what?

This name isn't descriptive enough of what we're testing. Perhaps ``test_that_flaky_works_with_top_level_functions``.

Comment (potrebic) [inline test/test_example.py]: In this example test module should there be more 'example' of this test function? Like tagging it flaky, etc?

File: flaky/flaky_nose_plugin.py [hunk 5]
```diff
@@ -152,27 +152,31 @@ def _get_test_method_name(test):
...
         :return:
-            The test method (and its name) that is being run by the test
+            The test callable (and its name) that is being run by the test
         :rtype:
             `tuple` of `callable`, `unicode`
         """
-        method_name = cls._get_test_method_name(test)
-        return getattr(test.test, method_name), method_name
+        callable_name = cls._get_test_callable_name(test)
+        if hasattr(test.test, callable_name):
```
Comment (potrebic) [inline flaky/flaky_nose_plugin.py]: How about this for L178-181

```
test_callable = getattr(test.test, callable_name, test.test)
```

File: flaky/flaky_nose_plugin.py [hunk 6]
```diff
@@ -152,27 +152,31 @@ def _get_test_method_name(test):
...
+            test_callable = getattr(test.test, callable_name)
+        else:
+            test_callable = test.test
+        return test_callable, callable_name
```
Comment (potrebic) [inline flaky/flaky_nose_plugin.py]: I don't see any unit test changes associated with this change in logic?

File: flaky/flaky_nose_plugin.py (hunk 5 above)
Comment (Jeff-Meadows) [inline flaky/flaky_nose_plugin.py]: Yes!

File: test/test_example.py (hunk 4 above)
Comment (Jeff-Meadows) [inline test/test_example.py]: Will add docstring.

File: flaky/flaky_pytest_plugin.py (hunk 3 above)
Comment (Jeff-Meadows) [inline flaky/flaky_pytest_plugin.py]: Should be. All the changes to the pytest side are just variable/parameter names. The bug was in nose only.

**Pull request merged**